### DISCUSSION TITLE
Annotating certain pages of a pdf

### DISCUSSION THREAD

**Discussion by pwesterbaan** [xournalpp/xournalpp#2572] [Q&A] [answered]
Is it possible to annotate a range of pages from a pdf? I'd like to use this software to teach from and I would like to annotate just the relevant pages for each day.

**Comment by rolandlo** [comment, marked as answer]
So you would like to have only the relevant pages loaded and displayed? This is not directly supported. You would have to extract the relevant pages with a tool like `pdftk`, annotate, and finally merge the resulting xopp-files using a script file like the one by @friciwolf, see issue #2474 (which you may need to modify to satisfy your needs).

**Reply by pwesterbaan** [reply]
I figured this was pretty much the only way. I've used pdftk before, but if I wanted to try and build an xournal++ plugin using pdftk, where can I find documentation on how to do that?

**Comment by rolandlo** [comment]
Our documentation for writing plugins is on the [work-in-progress-website](https://xournalpp.github.io/guide/plugins/). For using `pdftk` inside the plugin you would have to use the Lua `os.execute` or `io.popen` command. See [this site](https://simion.com/info/calling_external_programs.html) for documentation. You can also take a look at the [QuickScreenshot plugin](https://github.com/xournalpp/xournalpp/blob/master/plugins/QuickScreenshot/main.lua) how it is done.
